### PR TITLE
Added configuration options for resetting email.

### PR DESCRIPTION
**Pull request by pawel-brzezinski**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

Added the ability to configure options for sending e-mail with link to reset password to admin panel. Currently, it is not possible to configure the sender's email address (or message template). In many cases this causes problems with sending messages properly. Available options:

- token_send_limit - maximum number of email resend,
- sender - sender email address,
- subject - email subject (can be a phrase or translation key),
- template - view file with message template,
- translation_domain - the domain that should be used for translations in message template and for subject.

##### The following parameters are available:
- sulu_security.reset_password.mail.token_send_limit
- sulu_security.reset_password.mail.sender
- sulu_security.reset_password.mail.subject
- sulu_security.reset_password.mail.template
- sulu_security.reset_password.mail.translation_domain

##### The following variables are available in message template:
- user - user object,
- reset_url,
- translation_domain.

#### Example configuration
~~~yml
sulu_security:
    reset_password:
        mail:
            token_send_limit: 2
            sender: noreply@pbcms.pl
            subject: sulu_security.reset.password.email.subject
            template: ExampleBundle:mail_template:reset_password.html.twig
            translation_domain: example_domain
~~~

#### Default configuration
~~~yml
sulu_security:
    reset_password:
        mail:
            token_send_limit: 3
            # Default sender is generated as before
            sender: ~
            subject: security.reset.mail-subject
            # Default template is the same as before, just moved to the separate template file
            template: SuluSecurityBundle:mail_templates:reset_password.html.twig
            translation_domain: backend
~~~

#### BC Breaks/Deprecations

None

### CODE REVIEW

File: src/Sulu/Bundle/SecurityBundle/Controller/ResettingController.php [hunk 1]
```diff
@@ -143,34 +163,37 @@ protected function getTranslator()
...
+            UrlGeneratorInterface::ABSOLUTE_URL
         );
+        $template = $this->getParameter('sulu_security.reset_password.mail.template');
+
+        if (!$this->get('templating')->exists($template)) {
+            throw new EmailTemplateException($template);
+        }
 
-        return $message;
+        return trim($this->renderView($template, [
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Controller/ResettingController.php]: could you reformat this line?

```
trim(
    $this->renderView(
        $template,
        [...]
    )
);
```

File: src/Sulu/Bundle/SecurityBundle/Resources/views/mail_templates/reset_password.html.twig [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{{ 'security.reset.mail-message'|trans({}, translation_domain) }}
+{{ reset_url }}
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Resources/views/mail_templates/reset_password.html.twig]: missing last line

File: src/Sulu/Bundle/SecurityBundle/Security/Exception/EmailTemplateException.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    private $template;
+
+    /**
+     * EmailTemplateException constructor.
+     *
+     * @param string $template
+     */
+    public function __construct($template)
+    {
+        $this->template = $template;
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Security/Exception/EmailTemplateException.php]: call parent constructor first.

File: src/Sulu/Bundle/SecurityBundle/Controller/ResettingController.php [hunk 4]
```diff
@@ -126,7 +127,26 @@ public function resetAction(Request $request)
...
+    }
+
+    /**
+     * @param string $email
+     *
+     * @return bool
+     */
+    protected function isEmailValid($email)
+    {
+        $constraint = new EmailConstraint();
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Controller/ResettingController.php]: where do you have found this class? i cannot find this in my sulu installation.

Comment (wachterjohannes) [inline src/Sulu/Bundle/SecurityBundle/Controller/ResettingController.php]: ok i see (: you have used an as for class inclusion.

**Comment by wachterjohannes** [comment]
@pawel-brzezinski could you also update your branch? you can pull or rebase this branch from develop.

**Comment by pawel-brzezinski** [comment]
Hello @wachterjohannes,

> thanks for contribution. could you also add a line to the README?

To which README file should I add line? And what exactly should contain? Description for these new options?

Thanks,
Paweł

**Comment by wachterjohannes** [comment]
@pawel-brzezinski sorry (: I have meant the CHANGELOG file

**Comment by pawel-brzezinski** [comment]
@wachterjohannes, I added your requested changes. Branch is up-to-date with develop branch (with last commit on develop 7dba8e9). 
I'm not sure only what you mean in point with missing line in default e-mail template. I just added an empty line at the end of file (message looks exactly the same as before).

Thanks,
Paweł.

**Comment by wachterjohannes** [comment]
@pawel-brzezinski everything is ok (: that's what I want.

**Comment by wachterjohannes** [comment]
@pawel-brzezinski thanks for contribution

**Pull request merged**